alephium/contracts/sequence.ral

```diff
@@ -30,11 +30,16 @@ TxContract Sequence(
         next2 = 0
     }
 
-    fn checkSequence(seq: U256) -> () {
+    fn checkSequence(seq: U256) -> Bool {
         if (seq < next) {
             let undoneSequenceId = subContractId!(u256To8Byte!(seq / 256))
             UndoneSequence(undoneSequenceId).check(seq)
-            return
+            return true
+        }
+
+        if (seq >= (next + 768)) {
+            compact()
+            return false
         }
 
         if (seq >= (next + 512)) {
```

Comment:
Do we need a return for this branch?

Reply:
No, in this case, we will move the first window to the undone sequence contract and continue to verify the `seq`.

Reply:
I see! `next` is updated in `compact`